From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 404Setup <153366651+404Setup@users.noreply.github.com>
Date: Mon, 8 Jul 2024 19:48:30 +0800
Subject: [PATCH] Slice: Improve map-saving performance

Original code by Cryptite
You can find the original code on https://github.com/Cryptite/Slice

diff --git a/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java b/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
index 6fd0c8b12aef37e25d4a6bab553411fa790d79a6..e4160eb08dd5e9ce2ab29dbff8af9cfbdb9ac2bd 100644
--- a/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
+++ b/src/main/java/net/minecraft/world/level/saveddata/maps/MapItemSavedData.java
@@ -364,8 +364,8 @@ public class MapItemSavedData extends SavedData {
             --this.trackedDecorationCount;
         }
 
-        this.setDecorationsDirty();
         if (mapicon != null) {
+            this.setDecorationsDirty();
             if (mapicon.renderOnFrame()) this.dirtyFrameDecorations = true; // Paper
         }
     }
